lib/state.js

```diff
@@ -0,0 +1,9 @@
+import { eve } from './eve.js'
+
+// Create a frame observable that will update every frame
+const start = Date.now()
+export const frame = eve(start)
+requestAnimationFrame(function loop(tick) {
+  frame.set(start + tick)
+  requestAnimationFrame(loop)
+})
```

Comment:
this is where i loop over each frames to get an "active" tick system that's how we can detect changes on url, just check everytime if it has change, it's actually a pretty cheap operation